src/main/java/org/mskcc/cmo/metadb/service/impl/CmoLabelGeneratorServiceImpl.java

```diff
@@ -160,6 +159,12 @@ public String generateCmoSampleLabel(String requestId, IgoSampleManifest sampleM
 
         // get next incremement value for cmo sample counter
         Integer nextSampleCounter = getNextSampleIncrement(existingSamples);
+        // if the sample shares igoId/primaryId with a sample in existingSamples,
+        // this should reduce the counter by one or else remain as is
+        if (foundSampleInExistingSamples(sampleManifest.getIgoId(), existingSamples)) {
+            nextSampleCounter--;
```

Comment:
the increment should be the same if we find a matching sample by its igoId

if a list of existing samples returns with increments: 1, 2, 3, 6, 7 and the sample igo/primary id matches the primary id of the sample with increment 6 then we should use 6 as the increment

Reply:
Oh it needs to match the counter, not increase or decrease